### PR TITLE
refactor: replacing unwraps with errors (part 5) 

### PR DESCRIPTION
**Pull request by pompon0**
# What ❔

Moved unwraps outside of eth_sender_dal methods.
Removed unnecessary indentation from eth_sender_dal and contract_verification_dal. Renamed `Result<_,sqlx::Error>` to `sqlx::Result<_>`.

The final goal is to be able to safely enable panic=abort for the whole repo.

## Why ❔

Panics which do not terminate put the application into an invalid/unexpected state (which may cause state corruption for example) and can hide bugs which would be easy to notice if the application crashed immediately. Enabling panic=abort also incentives developers to make sure that the application is crash-safe.

### CODE REVIEW

File: core/lib/dal/src/eth_sender_dal.rs [hunk 1]
```diff
@@ -18,122 +19,108 @@ pub struct EthSenderDal<'a, 'c> {
...
-                        stats.saved.push((aggregation_action, batch_number));
-                    }
+                    .await?;
+
+            records.extend(
+                sqlx::query(&format!(
+                    "SELECT number as number, false as confirmed FROM l1_batches
+                 INNER JOIN eth_txs_history ON (l1_batches.eth_{}_id = eth_txs_history.eth_tx_id)
+                 ORDER BY number DESC
+                 LIMIT 1",
```
Comment (popzxc) [inline core/lib/dal/src/eth_sender_dal.rs:53]: Looks like the formatting has been broken here.

**Review by popzxc — APPROVED** [review]

File: core/lib/dal/src/contract_verification_dal.rs [hunk 2]
```diff
@@ -40,59 +39,55 @@ impl Display for Compiler {
...
-            .map(|row| row.count as usize)
-        }
+    pub async fn get_count_of_queued_verification_requests(&mut self) -> sqlx::Result<usize> {
+        sqlx::query!(
+            r#"
+            SELECT COUNT(*) as "count!"
+            FROM contract_verification_requests
+            WHERE status = 'queued'
+            "#
+        )
```
Comment (slowli) [inline core/lib/dal/src/contract_verification_dal.rs:49]: Nit: Since you're changing SQL query formatting from the `sqlx` perspective, it may make sense to apply "format-proofing" as in most newer queries (e.g., [here](https://github.com/matter-labs/zksync-era/blob/756ba6cbc5b0ecbcb336a6082ea61ab606f5fd82/core/lib/dal/src/blocks_dal.rs#L153)). This would make queries resistant to changes in the surrounding formatting and would make it easier to search queries in `sqlx-data.json`.

File: core/lib/dal/src/contract_verification_dal.rs [hunk 3]
```diff
@@ -181,238 +172,219 @@ impl ContractVerificationDal<'_, '_> {
...
+            status: row.status,
+            error: row.error,
+            compilation_errors: row
+                .compilation_errors
+                .and_then(|errors: serde_json::Value| {
+                    let string_array: Vec<String> = errors
+                        .as_array()
+                        .unwrap()
+                        .iter()
+                        .map(|value| value.as_str().unwrap().to_string())
```
Comment (slowli) [inline core/lib/dal/src/contract_verification_dal.rs:215]: Just to clarify: Do we consider it appropriate to `unwrap()` here and in similar places because we assume that the JSON shape will never get corrupted (e.g., by manually inserting a non-conforming value to the table)?

File: core/lib/dal/src/contract_verification_dal.rs [hunk 4]
```diff
@@ -181,238 +172,219 @@ impl ContractVerificationDal<'_, '_> {
...
+                        .collect();
+                    if string_array.is_empty() {
+                        None
+                    } else {
+                        Some(string_array)
+                    }
```
Comment (slowli) [inline core/lib/dal/src/contract_verification_dal.rs:221]: Nit / off-topic (?): Could be replaced with `(!string_array.is_empty()).then_some(string_array)`.

File: core/lib/zksync_core/src/api_server/web3/namespaces/debug.rs [hunk 5]
```diff
@@ -79,7 +79,8 @@ impl DebugNamespace {
         let call_trace = connection
             .blocks_web3_dal()
             .get_trace_for_miniblock(block_number)
-            .await;
+            .await
+            .unwrap();
```
Comment (slowli) [inline core/lib/zksync_core/src/api_server/web3/namespaces/debug.rs:83]: The error can be mapped here. Will this be done in a separate PR?

File: core/lib/zksync_core/src/eth_sender/tests.rs [hunk 6]
```diff
@@ -442,6 +448,7 @@ async fn three_scenarios() -> anyhow::Result<()> {
             .eth_sender_dal()
             .get_inflight_txs()
             .await
+            .unwrap()
```
Comment (slowli) [inline core/lib/zksync_core/src/eth_sender/tests.rs:451]: Nit: Since `eth_sender` tests return `anyhow::Result`, errors can be mapped here.

File: core/lib/dal/src/contract_verification_dal.rs (hunk 4 above)
Comment (pompon0) [reply core/lib/dal/src/contract_verification_dal.rs:221]: I don't find that readable

File: core/lib/dal/src/contract_verification_dal.rs (hunk 3 above)
Comment (pompon0) [reply core/lib/dal/src/contract_verification_dal.rs:215]: nah, I somehow missed these unwraps

File: core/lib/dal/src/eth_sender_dal.rs (hunk 1 above)
Comment (pompon0) [reply core/lib/dal/src/eth_sender_dal.rs:53]: done

File: core/lib/dal/src/contract_verification_dal.rs (hunk 2 above)
Comment (pompon0) [reply core/lib/dal/src/contract_verification_dal.rs:49]: done

File: core/lib/zksync_core/src/api_server/web3/namespaces/debug.rs (hunk 5 above)
Comment (pompon0) [reply core/lib/zksync_core/src/api_server/web3/namespaces/debug.rs:83]: I'm doing error propagation by layer-by-layer. Currently I'm just pushing out the panics out of the dal crate.

File: core/lib/zksync_core/src/eth_sender/tests.rs (hunk 6 above)
Comment (pompon0) [reply core/lib/zksync_core/src/eth_sender/tests.rs:451]: The error propagation in this tests seems somewhat inconsistent (there are both unwraps and returns). I'd rather lean towards standardizing on unwraps here (since it is a test), but it is out of scope of this pr.

File: core/lib/dal/src/contract_verification_dal.rs (hunk 3 above)
Comment (pompon0) [reply core/lib/dal/src/contract_verification_dal.rs:215]: fixed

File: core/lib/dal/src/eth_sender_dal.rs [hunk 7]
```diff
@@ -171,109 +153,109 @@ impl EthSenderDal<'_, '_> {
...
-            .execute(transaction.conn())
+    pub async fn confirm_tx(&mut self, tx_hash: H256, gas_used: U256) -> anyhow::Result<()> {
+        let mut transaction = self
+            .storage
+            .start_transaction()
             .await
-            .unwrap();
+            .context("start_transaction()")?;
+        let gas_used = i64::try_from(gas_used)
+            .map_err(|err| anyhow::anyhow!("Can't convert U256 to i64: {err}"))?;
```
[the remark this answers is not in the document]
Comment (pompon0) [reply core/lib/dal/src/eth_sender_dal.rs:214]: nope, I just blindly forward the error.

File: core/lib/dal/src/eth_sender_dal.rs [hunk 8]
```diff
@@ -18,122 +19,106 @@ pub struct EthSenderDal<'a, 'c> {
...
+    pub async fn get_inflight_txs(&mut self) -> sqlx::Result<Vec<EthTx>> {
+        let txs = sqlx::query_as!(
+            StorageEthTx,
+            "SELECT * FROM eth_txs WHERE confirmed_eth_tx_history_id IS NULL \
+             AND id <= (SELECT COALESCE(MAX(eth_tx_id), 0) FROM eth_txs_history WHERE sent_at_block IS NOT NULL) \
+             ORDER BY id"
+        )
+        .fetch_all(self.storage.conn())
+        .await?;
+        Ok(txs.into_iter().map(|tx| tx.into()).collect())
```
[the remark this answers is not in the document]
Comment (pompon0) [reply core/lib/dal/src/eth_sender_dal.rs:31]: sure, let's do that in a followup pr.

File: core/lib/dal/src/eth_sender_dal.rs (hunk 7 above)
[the remark this answers is not in the document]
Comment (pompon0) [reply core/lib/dal/src/eth_sender_dal.rs:214]: fixed

File: core/lib/dal/src/contract_verification_dal.rs [hunk 9]
```diff
@@ -181,247 +170,221 @@ impl ContractVerificationDal<'_, '_> {
...
+                LIMIT 1 \
+            ) storage_logs \
+            JOIN factory_deps ON factory_deps.bytecode_hash = storage_logs.value \
+            LEFT JOIN transactions ON transactions.hash = storage_logs.tx_hash \
+            WHERE storage_logs.value != $2",
+            hashed_key.as_bytes(),
+            FAILED_CONTRACT_DEPLOYMENT_BYTECODE_HASH.as_bytes()
+        )
+        .fetch_optional(self.storage.conn())
+        .await? else { return Ok(None) };
```
[the remark this answers is not in the document]
Comment (pompon0) [reply core/lib/dal/src/contract_verification_dal.rs:241]: ack

File: contracts
[the remark this answers is not in the document]
Comment (pompon0) [reply contracts:1]: done

**Review by slowli — APPROVED** [review]
Nice 👍
